src/emqx_protocol.erl

```diff
@@ -208,9 +208,21 @@ received(Packet = ?PACKET(Type), PState) ->
         true ->
             {Packet1, PState1} = preprocess_properties(Packet, PState),
             process_packet(Packet1, inc_stats(recv, Type, PState1));
-        {'EXIT', {topic_filters_invalid, _Stacktrace}} ->
+        {'EXIT', {protocol_error, _Stacktrace}, PState} ->
```

Comment:
As I commented in another PR.
Direct catch of an exception should consider deprecated and never use again.
The reason being:
unlike `try ... catch _ : _` 
`catch Expr` will always try to retrieve the stacktrace which is very expensive.
especially in our case here --- exception driven implementation,
stacktrace is always discarded, hence no need to retrieve at all.